internal/rollout/rollout.go

```diff
@@ -232,3 +248,23 @@ func (r *Rollout) diagnoseCandidate(candidate string, healthCriteria []config.He
 	d, err = health.Diagnose(ctx, healthCriteria, metricsValues)
 	return d, errors.Wrap(err, "failed to diagnose candidate's health")
 }
+
+func (r *Rollout) publish(svc *run.Service, diagnosis health.DiagnosisResult) error {
+	if r.pubsubClient == nil {
+		return nil
+	}
+
+	event, err := pubsub.NewRolloutEvent(svc, diagnosis, r.promoteToStable)
+	if err != nil {
+		return errors.Wrap(err, "failed to create message")
+	}
+	ctx := util.ContextWithLogger(r.ctx, r.log)
+	err = r.pubsubClient.Publish(ctx, event)
+	if err != nil {
+		return errors.Wrap(err, "error when publishing message")
+	}
+
+	// Wait for all messages to be sent (or to fail).
+	r.pubsubClient.Stop()
```

Comment:
Since we're publishing one event per go routine (in each rollout process), we might just wait for the event to be published using [`PublishResult.Get`](https://pkg.go.dev/cloud.google.com/go/pubsub?tab=doc#PublishResult.Get). In this way, we can abstract the synchronous publishing from the user of the pubsub package

Reply:
yeah that's a decent idea. I think that could be done with


    waitFn, err := PublishMessage()

Reply:
lets move Get into Publish